dgisim/src/agents.py

```diff
@@ -97,56 +96,55 @@ def choose_action(self, history: List[GameState], pid: GameState.Pid) -> PlayerA
                 else:
                     raise Exception("Game should end here but not implemented(NOT REACHED)")
                 
-            # food card
-            character_injured = active_character.get_hp() < active_character.get_max_hp()
+            # card action
             if selection < 1:
                 cards = me.get_hand_cards()
                 card: Optional[type[Card]]
-                tmp_dices = ActualDices({})
-
-                if cards.contains(SweetMadame) and character_injured:
-                    card = SweetMadame
-                    tmp_dices = ActualDices({Element.OMNI: 0})
-                elif cards.contains(MondstadtHashBrown) and character_injured:
-                    tmp_dices = ActualDices({Element.OMNI: 1})
-                    card = MondstadtHashBrown
-                elif cards.contains(MushroomPizza) and character_injured:
-                    tmp_dices = ActualDices({Element.OMNI: 1})
-                    card = MushroomPizza
-                else:
-                    card = None
-                
-                if card is not None and not active_character.satiated():
-                    print(card().name())
-                    return CardAction(
-                        card,
-                        CharacterTargetInstruction(
-                            tmp_dices,
-                            StaticTarget(
-                                pid,
-                                Zone.CHARACTER,
-                                active_character.get_id(),
+
+                # Consolidation Functions / Helpers
+                def try_card_action(card_class, condition, tmp_dices):
+                    if cards.contains(card_class) and condition():
+                        if active_character is not None:
+                            action = CardAction(
+                                card_class,
+                                CharacterTargetInstruction(
+                                    tmp_dices,
+                                    StaticTarget(
+                                        pid,
+                                        Zone.CHARACTER,
+                                        active_character.get_id()
+                                    )
+                                )
                             )
-                        )
-                    )
+                            if action is None:
+                                return None
+                            else:
+                                print(card_class().name())
+                                return action
+                        else:
+                            print("No Active Character")
+                            return None
```

Comment:
What I may do is just to assert at the beginning of the function that `active_character` is not `None` as you shouldn't really call the function when there's no active_character, don't handle it silently (print is quite silent).